packages/components/psammead-radio-schedule/src/utilities/index.test.js

```diff
@@ -0,0 +1,32 @@
+import detokenise from '@bbc/psammead-detokeniser';
+import durationDictionary from '.';
+
+describe('durationDictionary', () => {
+  it('creates a valid duration dictionary', () => {
+    expect(
+      durationDictionary({ duration: 'PT1H30M', locale: 'en-gb' }),
+    ).toEqual({
+      '%duration%': '1,30,00',
+    });
+  });
+  it('creates a valid duration dictionary and maps token to key correctly', () => {
+    expect(
+      detokenise(
+        'Duration %duration%',
+        durationDictionary({ duration: 'PT1H30M', locale: 'en-gb' }),
+      ),
+    ).toEqual('Duration 1,30,00');
+  });
+  it('creates a valid duration dictionary and returns with specified duration format', () => {
```

Comment:
What happens if duration and locale are invalid or missing?

Reply:
If duration is missing then I'd assume the fallback that radio schedules just wouldn't show would kick in before this fails. Are there any situations where we would get duration missing entirely? 
If locale was invalid or missing, again I'd assume other fallbacks at a higher level would come in first. Perhaps this needs further discussion elsewhere.

Reply:
We're discussing this at 2:30 today, we may want to render the card anyway even if we don't get back the duration. So we'll just have to make sure we're not trying to format it if it doesn't exist - let's make this decision based on the outcome of that meeting :)

Reply:
There should be fallbacks at a higher level, yes! My only concern is I think you mentioned that this threw an error if the duration isn't passed, or is passed in an incorrect format. Could we get a test case to cover what happens in this situation, and then we can decide if that's the behaviour we want, and take steps to guard against it if needed :)